### PR TITLE
Provision to use bespoke maintainers file through maintainers_file envVar

### PR DESCRIPTION
**Pull request by rgee0**
Signed-off-by: rgee0 <richard@technologee.co.uk>

Enables users to specify a maintainers file via a maintainers_file Env Var.

Also added a test for the new getEnv function and consolidated earlier tests into the same file, adding additional cases in the process.

### CODE REVIEW

File: commentHandler.go [hunk 1]
```diff
@@ -194,15 +196,19 @@ func isValidCommand(body string, trigger string) bool {
 
 func getMaintainers(owner string, repository string) []string {
 	client := http.Client{}
-	req, _ := http.NewRequest("GET", fmt.Sprintf("https://github.com/%s/%s/raw/master/MAINTAINERS", owner, repository), nil)
+
+	maintainersFile := getEnv(maintFileEnvName, defaultMaintFile)
```
Comment (alexellis) [inline commentHandler.go]: Looks good, but what about maintainersEnv?

Comment (rgee0) [reply commentHandler.go]: instead of `maintFileEnvName`?

Comment (alexellis) [reply commentHandler.go]: Yes or maintainersFileEnv

File: commentHandler.go [hunk 2]
```diff
@@ -194,15 +196,19 @@ func isValidCommand(body string, trigger string) bool {
...
+	maintainersFile = fmt.Sprintf("https://github.com/%s/%s/raw/master/%s", owner, repository, strings.Trim(maintainersFile, "/"))
+
+	req, _ := http.NewRequest("GET", maintainersFile, nil)
```
Comment (alexellis) [inline commentHandler.go]: How about? `http.MethodGet`

**Comment by rgee0** [comment]
@alexellis Good spot on the "GET" - should get into the habit of swapping those as & when the code is open

File: commentHandler.go [hunk 3]
```diff
@@ -194,15 +196,19 @@ func isValidCommand(body string, trigger string) bool {
 
 func getMaintainers(owner string, repository string) []string {
 	client := http.Client{}
-	req, _ := http.NewRequest("GET", fmt.Sprintf("https://github.com/%s/%s/raw/master/MAINTAINERS", owner, repository), nil)
+
+	maintainersFile := getEnv(maintainersFileEnv, defaultMaintFile)
+	maintainersFile = fmt.Sprintf("https://github.com/%s/%s/raw/master/%s", owner, repository, strings.Trim(maintainersFile, "/"))
```
Comment (alexellis) [inline commentHandler.go]: Maybe TrimLeft?

File: commentHandler.go [hunk 4]
```diff
@@ -194,15 +196,19 @@ func isValidCommand(body string, trigger string) bool {
 
 func getMaintainers(owner string, repository string) []string {
 	client := http.Client{}
```
Comment (alexellis) [inline commentHandler.go]: We need to implement a timeout value here for security.

File: commentHandler_test.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+		expectedAction: "",
+	},
+	{
+		title:          "Longer reopen command",
+		body:           "Derek reopen: ",
+		expectedAction: "reopen",
+	},
+	{
+		title:          "Longer close command",
+		body:           "Derek close: ",
```
Comment (alexellis) [inline commentHandler_test.go]: Is this new?

**Comment by alexellis** [comment]
Thank you for this contribution @rgee0. I'm merging also adding comments.

File: commentHandler_test.go (hunk 5 above)
Comment (rgee0) [reply commentHandler_test.go]: Its always been there as it drops through the pre-existing `isValidCommand` condition.

File: commentHandler.go (hunk 3 above)
Comment (rgee0) [reply commentHandler.go]: Used `trim` as it needs to be a file

**Pull request merged**